libnetdata/buffer/buffer.c

```diff
@@ -420,16 +427,22 @@ void buffer_increase(BUFFER *b, size_t free_size_required) {
     buffer_overflow_check(b);
 
     size_t left = b->size - b->len;
-
     if(left >= free_size_required) return;
 
-    size_t increase = free_size_required - left;
-    if(increase < WEB_DATA_LENGTH_INCREASE_STEP) increase = WEB_DATA_LENGTH_INCREASE_STEP;
+    size_t wanted = free_size_required - left;
+    size_t minimum = WEB_DATA_LENGTH_INCREASE_STEP;
+    if(minimum > wanted) wanted = minimum;
+
+    size_t optimal = b->size * 2;
```

Comment:
We will end up with 3x the memory that we had before the increase, right? This is a little bit too much. The amortized cost will be the same if we grow by a factor of 0.5x to 1x.

Reply:
this happens only for buffers < 1M

Reply:
I accepted your suggestion to go double the buffer size.